app/scienceapi/events/admin.py

```diff
@@ -1,5 +1,31 @@
 from django.contrib import admin
+from django.utils import timezone
+from django.utils.http import unquote
 
 from .models import Event
 
-admin.site.register(Event)
+
+@admin.register(Event)
+class EventAdmin(admin.ModelAdmin):
+
+    # Override add view so we can peek at the timezone they've entered and
+    # set the current time zone accordingly before the form is processed
+    def add_view(self, request, form_url='', extra_context=None):
+        if request.method == 'POST':
+            timezone.activate(request.POST.get('timezone'))
+        return super().add_view(request, form_url, extra_context)
+
+    # Override change view so we can peek at the timezone they've entered and
+    # set the current time zone accordingly before the form is processed
+    def change_view(self, request, object_id, form_url='', extra_context=None):
+        if request.method == 'POST':
+            timezone.activate(request.POST.get('timezone'))
+        else:
+            obj = self.get_object(request, unquote(object_id))
+            timezone.activate(obj.timezone)
+        return super().change_view(request, object_id, form_url, extra_context)
+
```

Comment:
So when a user gets the edit form the prepopulated time field should be in timezone user entered. I couldn't find a way to change the value of time field before rendering.